lib/src/configuration_values.dart

```diff
@@ -0,0 +1,148 @@
+      if (components.length < 3) {
+        continue;
+      }
+      if (components[0] != "--sentry-define") {
+        continue;
+      }
+      sentryArguments[components[1]] = components[2];
+    }
+    boolFromString(String? value) {
+      return value == "true"
+          ? true
+          : value == "false"
+              ? false
+              : null;
+    }
+
+    return ConfigurationValues(
+      version: sentryArguments['version'],
+      name: sentryArguments['name'],
+      uploadDebugSymbols: boolFromString(
+        sentryArguments['upload_debug_symbols'] ??
+            sentryArguments['upload_native_symbols'],
+      ),
+      uploadSourceMaps: boolFromString(sentryArguments['upload_source_maps']),
+      uploadSources: boolFromString(
+        sentryArguments['upload_sources'] ??
+            sentryArguments['include_native_sources'],
+      ),
+      project: sentryArguments['project'],
+      org: sentryArguments['org'],
+      authToken: sentryArguments['auth_token'],
+      url: sentryArguments['url'],
+      waitForProcessing: boolFromString(sentryArguments['wait_for_processing']),
+      logLevel: sentryArguments['log_level'],
+      release: sentryArguments['release'],
+      dist: sentryArguments['dist'],
+      webBuildPath: sentryArguments['web_build_path'],
+      commits: sentryArguments['commits'],
+      ignoreMissing: boolFromString(sentryArguments['ignore_missing']),
+    );
+  }
+
+  factory ConfigurationValues.fromReader(ConfigReader configReader) {
+    return ConfigurationValues(
+      version: configReader.getString('version'),
+      name: configReader.getString('name'),
+      uploadDebugSymbols: configReader.getBool(
+        'upload_debug_symbols',
+        deprecatedKey: 'upload_native_symbols',
+      ),
+      uploadSourceMaps: configReader.getBool('upload_source_maps'),
+      uploadSources: configReader.getBool(
+        'upload_sources',
+        deprecatedKey: 'include_native_sources',
+      ),
+      project: configReader.getString('project'),
+      org: configReader.getString('org'),
+      authToken: configReader.getString('auth_token'),
+      url: configReader.getString('url'),
+      waitForProcessing: configReader.getBool('wait_for_processing'),
+      logLevel: configReader.getString('log_level'),
+      release: configReader.getString('release'),
+      dist: configReader.getString('dist'),
+      webBuildPath: configReader.getString('web_build_path'),
+      commits: configReader.getString('commits'),
+      ignoreMissing: configReader.getBool('ignore_missing'),
+    );
+  }
+
+  factory ConfigurationValues.fromPlatformEnvironment(
+    Map<String, String> environment,
+  ) {
+    return ConfigurationValues(
+      release: environment['SENTRY_RELEASE'],
+      dist: environment['SENTRY_DIST'],
+    );
+  }
+
+  factory ConfigurationValues.merged(
+    ConfigurationValues args,
+    ConfigurationValues file,
+    ConfigurationValues platformEnv,
+  ) {
```

Comment:
```suggestion
  factory ConfigurationValues.merged({
    required ConfigurationValues args,
    required ConfigurationValues file,
    required ConfigurationValues platformEnv,
  }) {
```

I think using named parameters here would decrease possible positional errors since we're using 3 parametrs of the same type